pkg/apis/pipeline/v1/provenance.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+// Provenance contains all the information that needs to be recorded in a provenance i.e. ConfigSource
+type Provenance struct {
+	// ConfigSource identifies where the task/pipeline file came from.
+	ConfigSource *ConfigSource `json:"configSource,omitempty"`
+}
+
+// ConfigSource records where the task/pipeline file came from.
```

Comment:
```suggestion
// ConfigSource identifies the source where a resource came from.
// This can include Git repositories, Task Bundles, file checksums, or other information
// that allows users to identify where the resource came from and what version was used.
```

Reply:
Great doc for the field. Updated. Thanks @wlynch